ursa.ino

```diff
@@ -20,12 +20,12 @@ float PITCH_OFFSET_CHANGE = .999994;  // larger = pitchOffset changes slower
 float pitchOffset = -2.000;  // subtracted from the output in readMPU6050 so that zero pitch can correspond to balenced. Because the MPU6050 may not be mounted in the robot perfectly or because the robot's weight might not be perfectly centered, zero may not otherwise respond to perfectly balanced.
 
 // Set up the rgb led names; used to control common-anode RGB LEDs
-uint8_t RightledR = A15; //GPIO12
-uint8_t RightledG = A16; //GPIO14
-uint8_t RightledB = A17; //GPIO27
-uint8_t LeftledR = A10; //GPIO4
-uint8_t LeftledG = A11; //GPIO0
-uint8_t LeftledB = A12; //GPIO2
+uint8_t ledRightR = A15; // GPIO12
+uint8_t ledRightG = A16; // GPIO14
+uint8_t ledRightB = A17; // GPIO27
+uint8_t ledLeftR = A10; // GPIO4
+uint8_t ledLeftG = A11; // GPIO0
+uint8_t ledLeftB = A12; // GPIO2
```

Comment:
@dasso001 Where do these `A#` values come from? Why aren't they declared like the GPIO pins below? E.g.

```cpp
#define LED_RIGHT_R GPIO_NUM_12
... etc
```

Reply:
I'm not sure where you found those constants, but I think they just get replaced with the number after the GPIO in your comment. It's not harmful but is unnecessary. Try replacing A17 with 27 and so on.
Analog input pins are often referred to by an alias starting with A, but analog output pins are usually just called their standard pin name.
Actually, I think GPIO_NUM_# gets replaced by just # so can all that extra text be removed?

Reply:
I just copied the code from someone else who was controlling an RGB led, and it does work. The A values are the ADC numbers for those pins -- see the pinout on the second page at https://docs.google.com/presentation/d/1-IR1PcPpblGcg6qjzUVlXPURmsgIiwAIc1vXGE6-Q5M/edit#slide=id.g5ba977f02a_0_9. I'm not sure what it is SUPPOSED to be, but as I said, it does work as is.

Reply:
Huh okay. Seems like the compiler I'm using in Travis doesn't like it for some reason... but I can try to debug that later. I wonder if using the GPIO pin numbers would work instead.

Reply:
Here is where I saw how to do it (https://github.com/espressif/arduino-esp32/blob/master/libraries/ESP32/examples/AnalogOut/ledcWrite_RGB/ledcWrite_RGB.ino) -- maybe this will help?

Reply:
@dasso001 I noticed you said you're using ADC pin numbers. ADC=Analog to Digital Converter, or the analog input pins. It is a strange choice of that example to use the analog input alias numbers instead of just the pin numbers. Can you try switching to using the gpio numbers? if that still works, it would be clearer code and make Andrew's "code judger" happy, but if it breaks your code just stick with what works.

Reply:
Andrew will have to try it. Headbot is currently rooming with him, at OSU.

Reply:
That's great! @andrewda  I'm so glad you'll have a friend in college!😉